synapse/types/handlers/sliding_sync.py

```diff
@@ -703,7 +703,12 @@ def previously(last_token: T) -> "HaveSentRoom[T]":
 
     @staticmethod
     def never() -> "HaveSentRoom[T]":
-        return HaveSentRoom(HaveSentRoomFlag.NEVER, None)
+        # We use a singleton to avoid repeatedly instantiating new `never`
+        # values. The cast is redundant depending on `T`, hence the ignore.
+        return cast("HaveSentRoom[T]", _HAVE_SENT_ROOM_NEVER)  # type: ignore[redundant-cast]
```

Comment:
Something like this?

```suggestion
        # values. The cast is redundant because the token is always `None` for `NEVER` and doesn't depend on `T`, hence the ignore.
        return cast("HaveSentRoom[T]", _HAVE_SENT_ROOM_NEVER)  # type: ignore[redundant-cast]
```

Reply:
I realised that we could actually just declare `_HAVE_SENT_ROOM_NEVER` as `HaveSentRoom[Any]`, which will then work for all contexts and so we can remove the cast.

What was happening before is the first time `mypy` saw `_HAVE_SENT_ROOM_NEVER` it was in the context of a `HaveSentRoom[str]` and so it got type inferred to that.